### DISCUSSION TITLE
Potential bug: `messageUpsert` event triggering itself

### DISCUSSION THREAD

**Discussion by benalexb** [code-chat-br/whatsapp-api#135] [Q&A]
Hello everyone,

Maybe I am not using the webhook api as it was intended, but I am observing a weird behavior which leads to an infinite loop when trying to set up a flow where I am waiting for incoming messages and trying to reply to them right after. Am I missing something or is there an actual problem? I'm about to dive into the code to debug and potentially fix it, but I just wanted to make sure first that this is actually a problem.

### How to Reproduce
1. Start a new instance and connect to it.
2. Set up a webhook on this instance and only enable the `messageUpsert` event.
3. The expectation is that this event is fired only when messages arrive.

### Observed Behavior
* Receiving a message does trigger the event as expected.
* However, sending a message is also triggering this event.
* When setting up a flow where one is listening for incoming messages, and subsequently, replying to them, leads to an infinite loop as the `messageUpsert` event is being triggered for both sending and receiving messages.

Message Received (messageUpsert) -> Message Sent (also triggers messageUpsert) -> Loop starts

**Comment by jrCleber** [comment]
Goodnight!
All good!?

---

Then! This is the common behavior: every message received or sent is forwarded to the `messages.upsert` event. But, note that in the received message properties there is the `keyFromMe` property, which indicates whether or not a message was sent by the connected number.

- `keyFromMe=true`: message sent by the connected number
- `keyFromMe=false`: message received by the connected number

With this information you can filter the type of message for your automation flow.

---

```ts
if(message.KeyFromMe === false) {
  // follow the automation flow.
}
```

With this simple implementation you will ignore all messages sent by the connected number.

**Comment by benalexb** [comment]
@jrCleber oh man, I should have checked the body of the incoming request 🤦 thanks for info man! It's working perfectly!
